viewer/src/main/webapp/viewer-html/common/viewercontroller/ol/OlVectorLayer.js

```diff
@@ -169,14 +226,29 @@ Ext.define("viewer.viewercontroller.ol.OlVectorLayer", {
             me.idNumber++;
             me.stopDrawing();
         }, this);
-        
+        var keyDownListener = this.keyDown.bind(this);
+        var keyUpListener = this.keyUp.bind(this);
         this.polygon.on('drawend', function (evt) {
+            document.removeEventListener("keydown", keyDownListener, true);
+            document.removeEventListener("keyup", keyUpListener, true);
+            var coords  = evt.feature.getGeometry().getCoordinates();
+            coords[0].push(coords[0][0]);
+            evt.feature.getGeometry().setCoordinates(coords);
             me.select.setActive(true);
             evt.feature.setId("OpenLayers_Feature_Vector_" + me.idNumber);
             me.idNumber++;
             me.stopDrawing();
         }, this);
-        
+        this.polygon.on('drawstart', function (evt) {
+            // add key listeners
+            document.addEventListener("keydown", keyDownListener, true);
+            document.addEventListener("keyup", keyUpListener, true);
```

Comment:
misschien is het netter om niet aan `document` te binden voor de eventListener maar aan het DOM element van de OLMap... weet niet of je daar makkelijk bij kan of het dan nog werkt. misschien heeft @geertplaisier daar ideeen over

Reply:
Dit heb ik samen met Geert gemaakt maar misschien dat het nog beter kan.. @geertplaisier